src/diff/diffDirectories.test.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import _ from "lodash";
+import { expect } from "chai";
+import fs from "fs-extra";
+import path from "path";
+import sinon from "sinon";
+import tmp from "tmp";
+
+import * as diffProcessor from "./diffProcessor";
+import { RamlDiff } from "./diffProcessor";
+import { NodeDiff } from "./jsonDiff";
+import { diffNewAndArchivedRamlFiles } from "./diffDirectories";
+
+describe("diffNewAndArchivedRamlFiles", () => {
+  let leftDir: string;
+  let rightDir: string;
+  let leftApiConfigFile: string;
+  let rightApiConfigFile: string;
+
+  const apiConfig = {
+    family1: [
+      { assetId: "api1", fatRaml: { mainFile: "api1.raml" } },
+      { assetId: "api2", fatRaml: { mainFile: "api2.raml" } }
+    ],
+    family2: [
+      { assetId: "api3", fatRaml: { mainFile: "api3.raml" } },
+      { assetId: "api4", fatRaml: { mainFile: "api4.raml" } }
+    ]
+  };
+  const nodeDiffArr = [new NodeDiff("#/web-api/endpoints/test-endpoint")];
+  const diffRamlStub = sinon.stub(diffProcessor, "diffRaml");
+
+  beforeEach(() => {
+    leftDir = tmp.dirSync().name;
+    rightDir = tmp.dirSync().name;
+    leftApiConfigFile = path.join(leftDir, "api-config.json");
+    rightApiConfigFile = path.join(rightDir, "api-config.json");
+    fs.writeJsonSync(leftApiConfigFile, apiConfig);
+    fs.copyFileSync(leftApiConfigFile, rightApiConfigFile);
+    diffRamlStub.reset();
+    diffRamlStub.resolves(nodeDiffArr);
+  });
+
+  after(() => {
+    diffRamlStub.restore();
+  });
+
+  it("should return diff on all the apis in api-config.json", async () => {
+    const result: RamlDiff[] = await diffNewAndArchivedRamlFiles(
+      leftApiConfigFile,
+      rightApiConfigFile
+    );
+
+    require("fs-extra").writeJsonSync(path.join(__dirname, "tmp.json"), result);
+
+    expect(result.length).to.equal(4);
+    expect(diffRamlStub.callCount).to.equal(4);
+    expect(result[0].file).to.equal("api1/api1.raml");
```

Comment:
Does order if not might want to change these tests to be order agnostic in the results.  Would make for a more stable test.